modal/runner/engines/vllm.py

```diff
@@ -87,22 +88,29 @@ async def generate(self, payload: CompletionPayload, params):
                         and request_output.outputs[0].text[-1] == "\ufffd"
                     ):
                         continue
-                    token = request_output.outputs[0].text[index:]
-                    index = len(request_output.outputs[0].text)
-                    yield create_sse_data(token)
+
+                    final_output = request_output
+                    token = final_output.outputs[0].text[index:]
+                    index = len(final_output.outputs[0].text)
+                    yield create_sse_data(
+                        token,
+                        prompt_tokens=len(final_output.prompt_token_ids),
+                        completion_tokens=len(
+                            final_output.outputs[0].token_ids
+                        ),
+                        done=False,
+                    )
```

Comment:
previously we were sending 0 for token counts while streaming, this would now send a running total

OpenAI appears to not send any usage information in their streaming chunks? https://platform.openai.com/docs/api-reference/chat/object

but if the request stops short, we still want the info so might be better to send runnning totals

Reply:
unsure if the router repo will take running totals into consideration before the `done=True` flag is sent -- will need to double check

dont want accidental overbilling for streaming

Reply:
OpenAI doesn't do any accounting for streaming :(
We should, by sending the running token counts along with the running total time